modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/object/DefaultValueResolverParsingDelegate.java

```diff
@@ -32,11 +33,11 @@ public boolean accepts(MetadataType metadataType) {
   /**
    * @param key the parsed entity key
    * @param metadataType a {@link MetadataType}
-   * @param dslElementResolver
+   * @param elementDsl the {@link DslElementSyntax} of the parsed element
    * @return A {@link ValueResolver} that performs a registry lookup using the given {@code key}
    */
   @Override
   public ValueResolver<Object> parse(String key, MetadataType metadataType, DslElementSyntax elementDsl) {
-    return new RegistryLookupValueResolver<>(key);
+    return isBlank(key) ? new StaticValueResolver<>(key) : new RegistryLookupValueResolver(key);
```

Comment:
this should only apply if in tooling mode. If this happens in runtime then it's a bug and should blow up with a meaningful error